libs/design/tree/src/tree-item/tree-item.directive.ts

```diff
@@ -135,30 +138,51 @@ export class DaffTreeItemDirective {
   @HostListener('click')
   onClick() {
     if(this.node.hasChildren) {
-      this.toggleTree(this.node);
+      this.toggleTree();
     }
+  }
+
+  scrollIntoView() {
+    (<HTMLElement>this.elementRef.nativeElement).scrollIntoView();
+  }
+
+  /**
+   * Opens parent and parent of parent all the way to the root of the tree.
+   */
+  openAncestors() {
+    const openParent = (tree: DaffTreeUi<unknown>) => {
```

Comment:
Remove the recursion and replace with iteration.

Reply:
what list would be iterated? Recursion is naturally suited to tree structures

Reply:
I updated this to use `collect` (which is itself recursive but has the added protection of  a max depth) and iteration.